### PR TITLE
Adds console command to download and import a Rock The Vote report

### PR DESCRIPTION
**Pull request by aaronschachter**
### What's this PR do?

This pull request continues on RTV API integration (#126) by adding a `rtv:import {$reportId}` that downloads and imports the Rock The Vote report for a given ID, if its status is `complete`.

It also adds an `options` text field to the `import_files` table, to save and display any parameters passed when importing a file.

Example console output:

```

vagrant@homestead:~/Code/chompy$ php artisan rtv:import 1966
[2020-03-06 19:44:42] local.INFO: Checking status of report 1966  

[2020-03-06 19:44:42] local.INFO: Report 1966 status is complete  

[2020-03-06 19:44:45] local.INFO: Downloaded report 1966  

[2020-03-06 19:44:45] local.INFO: STARTING rock-the-vote IMPORT {"options":"Array
(
    [report_id] => 1966
)
"} 

[2020-03-06 19:44:45] local.INFO: Broadcasting [Chompy\Events\LogProgress] on channels [importer] with payload:
{
    "message": "Total rows to chomp: 1774",
    "type": "general",
    "progressValue": null
}  

```
Example import files:

Screen Shot 2020-03-06 at 11 54 26 AM

### How should this be reviewed?

👀 

### Any background context you want to provide?

In a future pull request(s), we'll make requests to the `POST /reports` endpoint in the RTV API to create a report and pass its ID to this command.

### Relevant tickets

References [Pivotal #171656723](https://www.pivotaltracker.com/story/show/171656723).

### Checklist

- [x] This PR has been added to the relevant Pivotal card.
- [ ] Added appropriate feature/unit tests.

### CODE REVIEW

File: app/Console/Commands/ImportRockTheVoteReport.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+     */
+    private function downloadAndImport($reportId, $url)
+    {
+        $path = 'uploads/' . ImportType::$rockTheVote . '-report-' . $reportId . '-' . Carbon::now() . '.csv';
+
+        Storage::put($path, app('Chompy\Services\RockTheVote')->getReportByUrl($url));
+
+        info('Downloaded report '.$reportId);
+
+        ImportFileRecords::dispatch(null, $path, ImportType::$rockTheVote, ['report_id' => $reportId])->delay(now()->addSeconds(3));
```
Comment (katiecrane) [inline app/Console/Commands/ImportRockTheVoteReport.php]: Why do we need to delay this by 3 seconds?

File: database/migrations/2020_03_06_000000_make_import_files_user_id_nullable.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('import_files', function (Blueprint $table) {
+            $table->string('user_id')->nullable()->change();
```
Comment (katiecrane) [inline database/migrations/2020_03_06_000000_make_import_files_user_id_nullable.php]: Why do we need this to be nullable?

File: app/Console/Commands/ImportRockTheVoteReport.php (hunk 1 above)
Comment (aaronschachter) [reply app/Console/Commands/ImportRockTheVoteReport.php]: That's a great question, I honestly don't know. I've been following by example per other jobs in the codebase, that all seem to have it. e.g. https://github.com/DoSomething/chompy/commit/884025eae0fcbab8acb6fbfc1fe7caf28f218a53#diff-e489ef4664b1c24151b9b01896103403

@DFurnes ?

Comment (katiecrane) [reply app/Console/Commands/ImportRockTheVoteReport.php]: oh lol I did it too! I don't remember why though hmmmm

File: database/migrations/2020_03_06_000000_make_import_files_user_id_nullable.php (hunk 2 above)
Comment (aaronschachter) [reply database/migrations/2020_03_06_000000_make_import_files_user_id_nullable.php]: When running this command from console, we want to insert a new `import_file` record, but there's no authenticated user that's uploading the file -- it's been done via the `rtv:import {reportId}` artisan command that this PR introduces.

File: app/Console/Commands/ImportRockTheVoteReport.php (hunk 1 above)
Comment (DFurnes) [reply app/Console/Commands/ImportRockTheVoteReport.php]: Yeah, I can't think of a good reason why we'd want to delay these.

File: database/migrations/2020_03_06_000000_make_import_files_user_id_nullable.php (hunk 2 above)
Comment (katiecrane) [reply database/migrations/2020_03_06_000000_make_import_files_user_id_nullable.php]: ah got it! thanks!

File: app/Console/Commands/ImportRockTheVoteReport.php (hunk 1 above)
Comment (katiecrane) [reply app/Console/Commands/ImportRockTheVoteReport.php]: Alright, your call then @aaronschachter !

Comment (aaronschachter) [reply app/Console/Commands/ImportRockTheVoteReport.php]: I'm going to err on the side of trusting the delays were added for a reason. I wonder if the broadcasting to Pusher could have anything to do with it. Let's saving testing with removing the delays for a separate PR.

**Pull request merged**